File: src/main/java/org/academy/api/client/gui/msdf/atlas/MsdfAtlasManager.java
package org.academy.api.client.gui.msdf.atlas;

import net.minecraft.resources.Identifier;
import org.academy.api.client.gui.msdf.Constants;

import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.Executor;

public final class MsdfAtlasManager {
    private static final ConcurrentHashMap<Identifier, MsdfAtlas> atlases = new ConcurrentHashMap<>();

    private MsdfAtlasManager() {
    }

    public static MsdfAtlas getAtlas(Identifier descriptor, Executor executor) {
        return atlases.computeIfAbsent(descriptor, _ -> new MsdfAtlas(
                Constants.DEFAULT_ATLAS_SIZE,
                Constants.DEFAULT_GLYPH_SIZE,
                Constants.DEFAULT_PX_RANGE,
                executor
        ));
    }

    public static void closeAll() {
        atlases.values().forEach(MsdfAtlas::close);
        atlases.clear();
    }
}

File: src/main/kotlin/org/academy/api/client/gui/msdf/atlas/MsdfAtlasManager.kt
package org.academy.api.client.gui.msdf.atlas

import net.minecraft.resources.Identifier
import org.academy.api.client.gui.msdf.core.Constants
import java.util.concurrent.ConcurrentHashMap

object MsdfAtlasManager {
    private val atlases: MutableMap<Identifier, MsdfAtlas> = ConcurrentHashMap<Identifier, MsdfAtlas>()

    fun getAtlas(descriptor: Identifier): MsdfAtlas {
        return atlases.computeIfAbsent(descriptor) { _ ->
            MsdfAtlas(
                Constants.DEFAULT_ATLAS_SIZE,
                Constants.DEFAULT_GLYPH_SIZE,
                Constants.DEFAULT_PX_RANGE
            )
        }
    }

    fun closeAll() {
        atlases.values.forEach { it.close() }
        atlases.clear()
    }
}